### PR TITLE
Correct tuple behaviour to correctly use pointers

### PR DESCRIPTION
**Pull request by EmilyBourne**
Fixes issue #328 

This branch should be ready to merge once pull request #334  is completed and merged here

### CODE REVIEW

**Comment by yguclu** [comment]
Could you change the title of this PR so that we have a meaningful message after squashing and merging?

File: pyccel/parser/semantic.py [hunk 1]
```diff
@@ -1030,51 +1032,55 @@ def _extract_indexed_from_var(self, var, args, name):
...
+        args = tuple(args)
+
         if isinstance(var, TupleVariable) and not var.is_homogeneous:
 
-            for i, arg in enumerate(args[::-1]):
-                if (not isinstance(arg, Integer) and
-                        not isinstance(arg, Slice)):
+            arg = args[-1]
+            if (not isinstance(arg, Integer) and
+                    not isinstance(arg, Slice)):
```
Comment (yguclu) [inline pyccel/parser/semantic.py]: Can't this be written as follows?
```
if not isinstance(arg, (Integer, Slice)):
    ...
```

Comment (EmilyBourne) [reply pyccel/parser/semantic.py]: This is outdated code. Are you looking at a specific commit rather than the final code?

Comment (yguclu) [reply pyccel/parser/semantic.py]: Sorry, that must have been the case

File: pyccel/parser/semantic.py [hunk 2]
```diff
@@ -1030,51 +1032,55 @@ def _extract_indexed_from_var(self, var, args, name):
...
+                if len(selected_vars)==1:
+                    if len(args) == 1:
+                        return selected_vars[0]
                     else:
-                        return PythonTuple(*[self._extract_indexed_from_var(var, args[:len(args)-i-1], name) for var in selected_vars])
-
+                        var = selected_vars[0]
+                        return self._extract_indexed_from_var(var, args[:-1], name)
+                elif len(selected_vars)<1:
+                    return None
```
[the remark this answers is not in the document]
Comment (yguclu) [reply pyccel/parser/semantic.py:1065]: That makes sense to me.

**Comment by EmilyBourne** [comment]
After adding the teardown function the tests have run twice without failures. I think we can assume that this fixes the problem. As a result this branch is now ready to merge

**Pull request merged**